### PR TITLE
[WIP] Testing out supporting distillation in open_clip

### PR DESCRIPTION
**Pull request by mitchellnw**
Can't support every feature so we may not want this, but just seeing how an initial experiment goes.

As per https://github.com/mlfoundations/open_clip/issues/370

This PR tries an alternative approach to @iejMac in https://github.com/iejMac/encoder-distill because it does not use MSE and instead just computes logits with both the student and teacher and does standard distillation loss

### CODE REVIEW

**Comment by mitchellnw** [comment]
All goes well distill OAI L/14 into B/32 will take 2.8 days https://wandb.ai/dogml/open_clip6/runs/distill_l14_32b_laion2b_v2

**Comment by mitchellnw** [comment]
final distillation number is 67.47% for B/32 (when distilling from L/14), so +4pp

**Comment by mitchellnw** [comment]
This got too confusing and I'm going to start again from scratch.
